### DISCUSSION TITLE
New Output: Current File -> Caret Position `line:column`

### DISCUSSION THREAD

**Discussion by FelipeRearden** [Taitava/obsidian-shellcommands#116] [Ideas]
Hello @Taitava !!!!

Following our conversation about Caret Position features in #114 

A New output to define a specific position in the Current Note to insert content. 

The initial idea is to have a new output with option to define the position in the note (line:column) where the content of the SC will be placed. 

Example: I want to insert the clipboard in the current note in the line 11 in the first caracter 

echo {{clipboard}} [Current File: Caret Position 6:0]

**Comment by Taitava** [comment]
I'd also add negative indexes:
- Line -1 is same as: insert on the last line (bottom). Line -2 is same as: insert on the second last line. And so on.
- Column -1 means to insert to the end of line. Column -2 means to insert before the last character on the line. And so on.

**Reply by FelipeRearden** [reply]
Fantastic idea @Taitava !!!!

Much better this way with negative values !!!!!!!!

**Comment by Taitava** [comment]
I'll quote an old output channel plan (your idea) here, that I haven't yet implemented:

https://github.com/Taitava/obsidian-shellcommands/discussions/16#discussion-3576047
> * Insert output to another note in background (without opening the destination note) + defined positions: 1- caret position , 2-top , 3-bottom (option to apply line break \n after/before 1 2 3) - Thanks @FelipeRearden !

The reasong why I haven't implemented this yet is that I haven't yet decided about the UI that is needed to provide extra configuration fields for output channels. Plus how to make the settings file structure to include field sets that depend upon the chosen output channel. Well, I'll figure something out for that. I think the UI part is actually quite well thought about already in the next quote:

https://github.com/Taitava/obsidian-shellcommands/discussions/16#discussioncomment-1452428
> I'm now thinking about creating fields that appear/hide whether they are needed or not, in the current _extra options modal_. It would be easier for a user when the fields will be accessible in the same modal as the _output channel_ selection.

TL;DR: All new output channels that require configuration parameters (be it a target file, a caret position or something else) are something that I do want to implement, but they will take more time.

**Reply by FelipeRearden** [reply]
Thanks for all the clarification @Taitava !!! 

This new outputs fits with our previous thoughts about new outputs 

Have a great day!
